cmd/serve_cmd.go

```diff
@@ -53,13 +54,43 @@ func serve(ctx context.Context) {
 	opts := []api.Option{
 		api.NewLimiterOptions(config),
 	}
-	a := api.NewAPIWithVersion(config, db, utilities.Version, opts...)
-	ah := reloader.NewAtomicHandler(a)
-	logrus.WithField("version", a.Version()).Infof("GoTrue API started on: %s", addr)
 
 	baseCtx, baseCancel := context.WithCancel(context.Background())
 	defer baseCancel()
 
+	var wg sync.WaitGroup
+	defer wg.Wait() // Do not return to caller until this goroutine is done.
+
+	if config.Worker.Enabled {
+		wrkLog := logrus.WithField("component", "workers")
+		wrk := worker.New(config, wrkLog)
+		opts = append(opts, &api.MailerOptions{
+			MailerClientFunc: wrk.GetMailerFunc,
+		})
+
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+
+			var err error
+			defer func() {
+				logFn := wrkLog.Info
+				if err != nil {
+					logFn = wrkLog.WithError(err).Error
+				}
+				logFn("worker is exiting")
```

Comment:
Maybe explain what worker? 
```suggestion
				logFn("background task worker is exiting")
```